### DISCUSSION TITLE
REST Query - Failed to send request (if request takes too long)

### DISCUSSION THREAD

**Discussion by tokkisva** [Budibase/budibase#14258] [Help] [answered]
Hi All,

**Hosting**
- Self
  - Method: docker-compose
  - Budibase Version: 2.1.42

**Describe the bug**
When running a REST query within the app, the request fails with message "Failed to send request". While the payload is not large, the time it takes for the query to run can be up to 30 seconds.

Updated QUERY_THREAD_TIMEOUT to 990000 under environment, did not help the issue.

When testing the call from the builder, the request is sent successfully. 
![image](https://github.com/user-attachments/assets/f6d65041-1078-4fee-aea8-c54e4ed02404)

Note: the GET request does not rely on input from the form, it is static and has no bindings. 

Error Logs: 

Browser - 
[Client] HTTP 400 on POST:/api/v2/queries/query_datasource_08fc21dc182741a087a281deeff0fa11_e2c07a2fb2f14437b7a4465d98233d00
	Failed to send request

Server -
bbproxy               | 2024/07/26 14:59:36 [info] 25#25: *13 epoll_wait() reported that client prematurely closed connection, so upstream connection is closed too while sending request to upstream, client: 172.19.0.1, server: _, request: "POST /api/v2/queries/query_datasource_08fc21dc182741a087a281deeff0fa11_e2c07a2fb2f14437b7a4465d98233d00 HTTP/1.1", upstream: "http://172.19.0.8:4002/api/v2/queries/query_datasource_08fc21dc182741a087a281deeff0fa11_e2c07a2fb2f14437b7a4465d98233d00", host: "_DELETED_", referrer: "https://_DELETED_/app/p2o-production-output

**Additional context**

Is there anything else I can try, or something that I'm missing? When I lower the amount of source data in the REST query, the query time shortens and the error goes away (but then I don't have the data needed in the app).

Thanks in advance!

**Comment by mike12345567** [comment, marked as answer]
Hi @tokkisva - this seems like you're running into an issue with the proxy (NGINX) - your Budibase version is quite out of date, with the latest version being `2.29.25`.

If you update your proxy service (image `budibase/proxy`) then this may have been solved already.

**Reply by R2bEEaton** [reply]
Hi, I was wondering - is it possible to update the proxy only and not all of Budibase? I would like to update the proxy for a similar issue to OP but I can't update my full Budibase install due to relying on deprecated features.

**Reply by mike12345567** [reply]
Hi @R2bEEaton - if you're using a docker-compose or Kubernetes installation then you can pull the latest `budibase/proxy` image and remove the container/restart the services, however if you're using the single image this isn't possible.
